Mathlib/Tactic/GCongr/Core.lean

```diff
@@ -521,9 +543,12 @@ elab_rules : tactic
     -- The core tactic `Lean.MVarId.gcongr` will be run with main-goal discharger being the tactic
     -- consisting of running `Lean.MVarId.gcongrForward` (trying a term together with limited
     -- forward-reasoning on that term) on each of the listed terms.
-    let assum g := g.gcongrForward hyps
+    let m : StateRefT Hypotheses MetaM (Bool × List (TSyntax ``binderIdent) × Array MVarId) := do
+      for h in hyps do
+        addHypothesis h
+      g.gcongr none [] (includeCreatedHyps := false)
```

Comment:
This is slightly different from the `addHypothesis` call above in that it omits the `isImplementationDetail`: is that intentional? Maybe extract the common loop so the difference stands out.